.github/workflows/test.yaml

```diff
@@ -104,3 +104,66 @@ jobs:
       - name: Test
         shell: bash
         run: ci/scripts/test.sh $(pwd)
+
+  integration:
+    name: AMD64 integration
```

Comment:
I am wondering whether `AMD64` is relevant at the moment, I just feel it visually clutters the jobs and makes it slightly more difficult to find what I am looking for:
![image](https://github.com/user-attachments/assets/2090d059-1756-47e4-8eb0-f93102a68577)

We use uppercase on apache/arrow-go (`AMD64 Conda Integration`) and apache/arrow (`Integration / AMD64 Conda Integration Test`). We use lowercase on apache/arrow-java. I prefer uppercase, easier to read the important word

```suggestion
    name: AMD64 Integration
```

Reply:
Good point! I'll remove "AMD64" from all job names, change "integration" to "Integration" and merge this.